tracing/tracer.go

```diff
@@ -206,7 +206,7 @@ func (t *Tracer) StartSpan(operationName string, opts ...opentracing.StartSpanOp
 	} else {
 		span.trace.traceID = t.newTraceID()
 		span.trace.sampled = randbp.ShouldSampleWithRate(t.sampleRate)
-		initRootSpan(span)
+		initRootSpan(context.Background(), span)
```

Comment:
🔕 I think the context should be plumbed in explicitly since the rule of thumb is "Background only in main() and TestX"

Reply:
what do you mean by "plumbed in explicitly"?

here we are just using the context object in order to reuse the code. basically in `initRootSpan` we call `log.Attach` to create the logger and sentry hub, then we discard the logger (because we don't use it) and get the created hub out of the ctx and attach that to the span. so we need a parent context here (because we should never pass along nil contexts).

Reply:
🔕 the context should generally be passed in, rather than assuming context.Background